Automattic/jetpack
# Add Payment Buttons block

Fixes https://github.com/Automattic/jetpack/issues/24767

#### Changes proposed in this Pull Request:

This PR adds a new Payment Buttons block (`jetpack/payment-buttons`) that acts as a container of Payment Button blocks (`jetpack/recurring-payments`), effectively allowing users to add multiple payment buttons inline.

<img width="1008" alt="Screen Shot 2022-07-27 at 16 42 43" src="https://user-images.githubusercontent.com/1233880/181278059-8c381acf-3830-4c1e-a03e-56520b772a31.png">

The approach follows the same behavior as the Buttons block from Core (`core/buttons`), so the individual Payment Button block doesn't show up in the block inserter and can be only added as a child of the Payment Buttons block (existing Payment Button blocks are not affected and should keep working).

#### Other information:


#### Jetpack product discussion
N/A.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
- [Spin up a Jurassic Ninja site running this branch](https://jurassic.ninja/create/?jetpack-beta&branches.jetpack=add/payment-buttons-block).
- Set up Jetpack and start with a Free plan.
- Go to `/wp-admin/post-new.php`.
- Make sure you cannot insert Payment Button block in the post.
- Insert a Payment Buttons block.
- Make sure you're asked to upgrade.
- Make sure the upgrade banner is not get duplicated when selecting the inner blocks.
- Upgrade your plan.
- Make sure you're asked to connect to Stripe.
- Make sure the Stripe nudge and toolbar are not duplicated when selecting the inner blocks.
- Connect to Stripe (check PCYsg-lW4-p2#sandbox-method if you want to avoid entering real data).
- Select the container block.
- Make sure you can add multiple Payment Button blocks.
- Make sure you can set different plans on all of them.
- Make sure you can change the button width for all of them.
- Publish the post.
- Make sure the different payment buttons are visible as expected.
